### ISSUE TITLE
UserPoolTokens are null when in running in release

### ISSUE THREAD

**Issue by momo1** [aws-amplify/amplify-flutter#286]
The `CognitoAuthSession.userPoolTokens` are not populated when the APK is built in release mode. It works fine when built with debug or profile modes.

**To Reproduce**
```
CognitoAuthSession session = await Amplify.Auth.fetchAuthSession(options: CognitoSessionOptions(getAWSCredentials: true));
print("Access token is ${session.userPoolTokens.accessToken}");
```
Build and install: `flutter build apk --release && adb install build/app/outputs/flutter-apk/app-release.apk`

**Expected behavior**
I would expect it to work in release just like it does in debug and profile modes which output: ```Access token is eyJraWQiOiJkN3ZoT0tkNn....```

**Actual behavior**
In release mode: ```flutter : [ERROR:flutter/lib/ui/ui_dart_state.cc(177)] Unhandled Exception: NoSuchMethodError: The getter 'accessToken' was called on null.```

**Platform**
[X] Android
[] iOS <-- did not try yet

**Smartphone (please complete the following information):**
 - Device: LG G7 physical phone, Samsung S8 physical phone, and the Pixel 2 emulator 
 - OS: Android 9
 - Browser N/A
 - Version

**Comment by jamesonwilliams** [comment]
`NoSuchMethodError` is different than a regular `NullPointerException`. One thing to look for: perhaps the R8 shrinker is removing something that it shouldn't be. Can probably be fixed an update to the ProGuard rules file. Without having access to the rest of your error log, maybe `com.amazonaws.mobile.client.results.Tokens` or `com.amplifyframework.auth.cognito.AWSCognitoUserPoolTokens` is absent.

**Comment by momo1** [comment]
Thanks! It was indeed R8. I made a proguard file with `-printusage` and found that R8 is removing a ton of stuff from `com.amplifyframework.api.aws`. I now have a `proguard-rules.pro` file that looks like this:

```
-keep class com.amazonaws.** { *; }
-keep class com.amplifyframework.** { *; }
```
The `-keep class com.amazonaws.** { *; }` is probably not needed but I added it to be on the safe side. It barely makes a dent in the overall app size.
